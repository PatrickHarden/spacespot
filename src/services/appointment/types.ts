export interface AppointmentState {
  data?: Array<Appointment>
  error?: string
}

export interface Appointment {
  id: number
  enquireId: number
  description: string
  peer1Id: string
  peer2Id: string
  proposedTime: string
  createdAt: string
  lastUpdatedAt: string
  lastProposedBy: string
  appointmentStatus: string
}

export interface NewAppointment {
  description: string
  proposedTime: string
}
export interface EditAppointment {
  id: string
  description: string
  proposedTime: string
}
export interface AppointmentAction {
  type: string
  payload?: Array<Appointment> | NewAppointment | EditAppointment | string
}
