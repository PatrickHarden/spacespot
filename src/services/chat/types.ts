export interface ChatAction {
  type: string
  payload?: ChatResponse | ChatMessage | string
}
export interface MessageAction {
  type: string
  payload?: string
}
export interface ChatState {
  data?: ChatResponse
  error?: string
}

export interface ChatResponse {
  chatId: number
  enquiryId: number
  peer1Id: string
  peer2Id: string
  messages: Array<ChatMessage>
}

export interface ChatMessage {
  userId: string
  message: string
  sentAt: string
}
