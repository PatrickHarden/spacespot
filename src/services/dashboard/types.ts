import { Space } from 'services/space/types'

export interface DashboardState {
  buildings: Array<Space>
  spaces: { [key: string]: Array<Space> }
  error?: string
  isLoading: boolean
  initOK: boolean
  pendingDelete: Array<string>
}
