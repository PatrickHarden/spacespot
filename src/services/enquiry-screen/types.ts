export interface EnquiryScreenAction {
  type: string
  payload: EnquiryScreenStateData | string
}
export interface EnquiryScreenState {
  data?: EnquiryScreenStateData
  error?: string
}
export interface EnquiryScreenStateData {
  id: string
}
