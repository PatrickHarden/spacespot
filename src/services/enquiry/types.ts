import { FileItem } from 'services/global/types'
import { NegotiationTerm } from 'services/negotiation/types'

export interface DocumentDTO {
  fileId: number
  fileName: string
  enquireId: number
  uploadAt: Date
  uploadedBy: string
  documentType?: string
}

export const DOC_TYPE_TEMPLATE = 'CompanyTemplate'

export type DocData = any

export interface EnquiryAction {
  type: string
  payload:
    | string
    | number
    | DocumentDTO
    | Array<DocumentDTO>
    | EnquiryResponse
    | FileItem
    | DocData
    | EnquiryRequest
}

export interface EnquiryState {
  data?: EnquiryResponse
  docs?: Array<DocumentDTO>
  error?: string
}

export enum EnquiryType {
  HotDesk = 'HotDesk',
  FixedDesk = 'FixedDesk',
  ServicedOffice = 'ServicedOffice',
  notChecked = '',
}

export interface EnquiryResponse {
  chatId: number
  enquireId: number
  spaceId: string
  /* Tenant */
  peer1Id: string
  /* Landlord */
  peer2Id: string
  negotiationId?: number
  enquireStatus?: string
  flexType?: EnquiryType
  flex: boolean
}

export interface EnquiryAPIResultActions {
  successSignInAfter?: boolean
  errorShowMessage?: string
}

export interface EnquiryRequest {
  spaceId: string
  message: string
  negotiation: {
    flexible: boolean
    terms: Array<NegotiationTerm>
  }
  userId?: string
  loggedIn?: boolean
  apiResultActions?: EnquiryAPIResultActions
}

export interface EnquiryRequestFlex {
  spaceId: string
  message: string
  flex: boolean
  flexType: EnquiryType
  flexIdentifier?: string
  negotiation: {
    terms: Array<NegotiationTerm>
  }
  userId?: string
  loggedIn?: boolean
  apiResultActions?: EnquiryAPIResultActions
}
