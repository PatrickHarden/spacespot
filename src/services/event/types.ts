export interface EnquiryEvent {
  eventId: number
  eventMessage: any
  eventType: string
  eventDate: Date
}

export interface EnquiryEventState {
  [enquiryId: string]: {
    events: Array<EnquiryEvent>
    isLoading?: boolean
    error?: string
  }
}

export interface GetEnquiryEventsAction {
  type: string
  payload: {
    enquiryId: string
  }
}

export interface GetEnquiryEventsSuccessAction {
  type: string
  payload: {
    enquiryId: string
    events: Array<EnquiryEvent>
  }
}

export interface GetEnquiryEventsErrorAction {
  type: string
  payload: {
    enquiryId: string
    error: string
  }
}

export type EnquiryEventAction =
  | GetEnquiryEventsAction
  | GetEnquiryEventsSuccessAction
  | GetEnquiryEventsErrorAction
