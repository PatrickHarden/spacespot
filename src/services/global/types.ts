import { ChatState } from 'services/chat/types'
import { EnquiryState } from 'services/enquiry/types'
import { EnquiryScreenState } from 'services/enquiry-screen/types'
import { SpaceState } from 'services/space/types'
import { UserState } from 'services/user/types'
import { AppointmentState } from 'services/appointment/types'
import { OnBoardingState } from 'services/onboarding/types'
import { NegotiationState } from 'services/negotiation/types'
import { DashboardState } from 'services/dashboard/types'
import { MyEnquiriesState } from 'services/myEnquiries/types'
import { SearchState } from 'services/search/types'

export interface SpacespotState {
  chat?: ChatState
  dashboard?: DashboardState
  enquiry?: EnquiryState
  enquiryscreen?: EnquiryScreenState
  space?: SpaceState
  user?: UserState
  appointment?: AppointmentState
  onboarding?: OnBoardingState
  negotiation?: NegotiationState
  myenquiries?: MyEnquiriesState
  search?: SearchState
}

export interface SpacespotErrorApi {
  error?: any
  message?: any
}

export interface FileItem {
  key: number
  uri: string
  name: string
  type: string
}
