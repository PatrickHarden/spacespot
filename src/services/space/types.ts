import { SpacespotErrorApi } from 'services/global/types'
import {
  PropertyListing,
  CommonPhotos1,
  CommonBrochures1,
} from './PropertyListingschema'
import { DropItem } from 'components/OnBoarding/DropZone'
import { SpaceStatus, AmenityOptions } from 'services/onboarding/types'

export type Space = PropertyListing

// SpaceSpot types
export interface SearchOptions {
  region: string
  locale: string
  landLord?: string
  isParent?: boolean
  parentProperty?: string
}

export interface FilterOptions {
  filters: FilterState
  locale: string
  isParent?: boolean
}

export type FilterState = {
  [key: string]: string | Array<string>
}

export interface SpaceState {
  search?: SpaceSearchResponse | SpacespotErrorApi
  selectedSpace?: SpaceGetResponse
  parentSpace?: SpaceGetResponse
  regionAmenities?: AmenityOptions
  item?: SpaceGetResponse | SpacespotErrorApi
  items?: { [key: string]: Space | SpacespotErrorApi }
}

export interface SpaceGetResponse {
  Found: boolean
  ElapsedTime: string
  Document: Space
}

export interface SpaceSearchResponse {
  Found: boolean
  DocumentCount: number
  Documents: Array<Array<Space>>
  Document?: Space
  Took: string
}

export interface SpaceAction {
  type: string
  payload?:
    | SpaceGetResponse
    | SpaceSearchResponse
    | Space
    | SearchOptions
    | string
    | AmenityOptions
    | FilterOptions
    | FilterState
}

export interface TranslatePropSpace {
  images?: Array<DropItem>
  photo?: string
  key: string
  buildingId?: string
  name: string
  address: string
  city: string
  postCode: string
  country: string
  description: string
  location?: string
  floor: number
  latLng: { lat: number; lng: number }
  district: string
  subdistrict: {
    id: number
    name: string
  }
  type: string
  size: number
  rent: string
  availability?: Date
  aspects: any
  isNew?: boolean
  status: SpaceStatus
  hotDesks: number
  fixedDesks: number
  sevicedOffices: number
  uploadedImages?: CommonPhotos1
  uploadedDocs?: CommonBrochures1
}
