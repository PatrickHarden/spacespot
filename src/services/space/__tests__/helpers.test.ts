import { get } from 'lodash'
import { Space } from '../types'
import {
  generateGallery,
  getCharges,
  getImages,
  getFloorPlans,
} from '../helpers'

import Space1 from '../__mocks__/space1.json'
import dashboardBuilding from '../__mocks__/dashboard-building-space.json'
import { DashboardState } from 'services/dashboard/types'
import { SpacespotState } from 'services/global/types'

describe('Space helpers', () => {
  const spaceDocument1: Space = {
    ...((Space1 as unknown) as Space),
  }
  const dashboardBuildingDocument: SpacespotState = {
    ...((dashboardBuilding as unknown) as SpacespotState),
  }

  it('selector should return floorplans', () => {
    expect(getFloorPlans(spaceDocument1)).toMatchSnapshot()
  })
  it('selector should return getImages', () => {
    expect(getImages(spaceDocument1)).toMatchSnapshot()
  })
  it('selector should return getCharges of space Document1', () => {
    expect(getCharges(spaceDocument1)).toMatchSnapshot()
  })
  it('selector should return getCharges of dashboard building', () => {
    expect(
      getCharges(
        ((dashboardBuildingDocument.dashboard as unknown) as DashboardState)
          .spaces['SS_BUILDING_325'][0] as Space,
      ),
    ).toMatchSnapshot()
  })
  it('selector should return getCharges of space Document1', () => {
    expect(
      generateGallery(get(spaceDocument1, '["Common.Photos"][0]')),
    ).toMatchSnapshot()
  })
})
