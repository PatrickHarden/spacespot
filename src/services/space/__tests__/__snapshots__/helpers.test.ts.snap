// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space helpers selector should return floorplans 1`] = `
Array [
  Object {
    "original": "https://uat-search.cbrelistings.com/resources/fileassets/SS_SPACE_648/6862dd87/fplan1.jpg",
    "thumbnail": "https://uat-search.cbrelistings.com/resources/fileassets/SS_SPACE_648/6862dd87/fplan1.jpg",
  },
]
`;

exports[`Space helpers selector should return getCharges of dashboard building 1`] = `
Object {
  "additionalCosts": 0,
  "currencyCode": "NOK",
  "rent": 21,
  "serviceCharges": 1200,
  "total": 1221,
}
`;

exports[`Space helpers selector should return getCharges of space Document1 1`] = `
Object {
  "additionalCosts": 0,
  "currencyCode": "EUR",
  "rent": 0,
  "serviceCharges": 0,
  "total": 0,
}
`;

exports[`Space helpers selector should return getCharges of space Document1 2`] = `
Object {
  "alt": "photo1.jpg",
  "url": "https://uatlistingssearchcbreeun.blob.core.windows.net/fileassets/SS_SPACE_648/b904b44f/photo1.jpg",
}
`;

exports[`Space helpers selector should return getImages 1`] = `
Array [
  Object {
    "original": "https://uatlistingssearchcbreeun.blob.core.windows.net/fileassets/SS_SPACE_648/b904b44f/photo1.jpg",
    "thumbnail": "https://uatlistingssearchcbreeun.blob.core.windows.net/fileassets/SS_SPACE_648/b904b44f/photo1.jpg",
  },
]
`;
